src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,454 @@
+ */
+contract RwaMultiSwapOutputConduit {
+    /// @notice DAI token contract address.
+    DaiAbstract public immutable dai;
+
+    /// @notice Addresses with admin access on this contract. `wards[usr]`
+    mapping(address => uint256) public wards;
+    /// @dev Addresses with operator access on this contract. `can[usr]`
+    mapping(address => uint256) public can;
+    /// @notice Whitelist for addresses which can be picked. `bud[who]`
+    mapping(address => uint256) public bud;
+    /// @notice Addresses with push access on this contract. `may[usr]`
+    mapping(address => uint256) public may;
+    /// @notice PSM addresses whitelist. `pal[psm]`
+    mapping(address => uint256) public pal;
+
+    /// @notice PSM contract address.
+    address public psm;
+    /// @notice GEM Recipient address.
+    address public to;
+    /// @notice Destination address for DAI after calling `quit`.
+    address public quitTo;
+
+    /**
+     * @notice `usr` was granted admin access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` admin access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice `usr` was granted operator access.
+     * @param usr The user address.
+     */
+    event Hope(address indexed usr);
+    /**
+     * @notice `usr` operator access was revoked.
+     * @param usr The user address.
+     */
+    event Nope(address indexed usr);
+    /**
+     * @notice `usr` was granted push access.
+     * @param usr The user address.
+     */
+    event Mate(address indexed usr);
+    /**
+     * @notice `usr` push access was revoked.
+     * @param usr The user address.
+     */
+    event Hate(address indexed usr);
+    /**
+     * @notice `who` address whitelisted for pick.
+     * @param who The user address.
+     */
+    event Kiss(address indexed who);
+    /**
+     * @notice `who` address was removed from whitelist.
+     * @param who The user address.
+     */
+    event Diss(address indexed who);
+    /**
+     * @notice `psm` address whitelisted for hook.
+     * @param psm The PSM address.
+     */
+    event Clap(address indexed psm);
+    /**
+     * @notice `psm` address was removed from whitelist.
+     * @param psm The user address.
+     */
+    event Slap(address indexed psm);
+    /**
+     * @notice `psm` address was choosen.
+     * @param psm The PSM address.
+     */
+    event Hook(address indexed psm);
+    /**
+     * @notice `who` address was picked as the recipient.
+     * @param who The user address.
+     */
+    event Pick(address indexed who);
+    /**
+     * @notice `amt` amount of GEM was pushed to the recipient `to` using `psm`.
+     * @param psm PSM address used for swap.
+     * @param to Destination address for GEM.
+     * @param amt The amount of GEM.
+     */
+    event Push(address indexed psm, address indexed to, uint256 amt);
```

Comment:
I'm thinking whether `gem` is more relevant to this event than `psm`.
Having `gem` allows for easily querying push events for a specific stablecoin very effectively.

Reply:
Maybe we should have both:
```suggestion
    /**
     * @notice `amt` amount of GEM was pushed to the recipient `to` using `psm`.
     * @param psm PSM address used for swap.
     * @param to Destination address for GEM.
     * @param amt The amount of GEM.
     */
    event Push(address indexed psm, address indexed gem, address indexed to, uint256 amt);
```